src/counter.cpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * Source code for Urban motor controller encoder functionality
+ */
+
+#include "../include/main.h"
+
+// According to the AMT103 datasheet, the encoder is preset to PPR=2048 (meaning that CPR=8192)
+//  Encoder(int encA, int encB , int cpr, int index)
+//  - encA, encB    - encoder A and B pins
+//  - ppr           - impulses per rotation  (cpr=ppr*4)
+//  - index pin     - (optional input)
+Encoder encoder = Encoder(15, 4, 8192);
```

Comment:
Not for this PR, but for future, I believe the optical encoder you're using does have an index pin. You could potentially make use of it. But all it would provide is absolute position, you don't need it for relative position.

Reply:
Yes, @raulvazquezg will know and we'll update this in a future PR. There will likely be 1-2 more PRs for setting all pins and parameters up.

Reply:
That is correct. We did not include an index pin since the beginning. We will be doing velocity control testing this upcoming Saturday and @valentino-jaber and I will determine if it is fully necessary for the most optimal implementation.